internal/bminventory/inventory.go

```diff
@@ -2787,31 +2790,35 @@ func (b *bareMetalInventory) calculateHostNetworks(log logrus.FieldLogger, clust
 func (b *bareMetalInventory) listClustersInternal(ctx context.Context, params installer.V2ListClustersParams) ([]*models.Cluster, error) {
 	log := logutil.FromContext(ctx, b.log)
 	db := b.db
+
+	var dbClusters []*common.Cluster
+	var clusters []*models.Cluster
+
+	//this call should be before OwnedBy
+	if params.Owner != nil {
```

Comment:
maybe worth adding it to the OwnedBy API? i.e. could be cleaner as a simple arg in signature.

Reply:
We should keep the API as generic as possible. I am removing this assignment based on the design discussion with @avishayt